### DISCUSSION TITLE
Constrained BO to Solve Minimization Problem

### DISCUSSION THREAD

**Discussion by jan-99** [meta-pytorch/botorch#1074] [Q&A]
Hi,

My problem is similar to this https://github.com/pytorch/botorch/issues/719 where I'm trying to maximize -f(x) in order to minimize f(x) (where f(x) > 0 for all x).  I've tried setting the `infeasible_cost` argument of `ConstrainedMCObjective` as per https://github.com/pytorch/botorch/issues/719#issuecomment-785692589. 

I set up my optimization problem similar to [this tutorial](https://botorch.org/tutorials/closed_loop_botorch_only) (Closed-loop batch, constrained BO in BoTorch with qEI and qNEI) and with the `infeasible_cost` arg, but I am getting -0.0 for my optimized objective values. I think it is due to the way the `best_observed_value` is determined:

```
best_observed_value = weighted_obj(train_x).max().item()
```
```
def weighted_obj(X):
    """Feasibility weighted objective; zero if not feasible."""
    feasible = (outcome_constraint_1(X) <= 0)     
    return objective_function(X) * feasible.type_as(X)
```
The infeasible points that are set to 0 end up being chosen as the best observed value instead. May I know how `best_observed_value` should be determined for such a minimization problem?

Also, would it be possible to perform constrained BO for a minimization function where the known minima is 0.0? Since the infeasible points are set to 0, it seems that the BO would think that the true minima is an infeasible point.

**Comment by Balandat** [comment]
The approach we use for feasibility-weighting essentially assumes that the function you're optimizing is positive, which doesn't seem to hold in your case (you're optimizing `-f` with `f` positive). Have you tried the strategy discussed [here](https://github.com/pytorch/botorch/blob/3318c06333166328c95d24edd1055d4ef4bded70/botorch/utils/objective.py#L125-L129)?
